solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -562,8 +566,36 @@ public static AssignStrategy createAssignStrategy(CoreContainer coreContainer) {
         coreContainer.getPlacementPluginFactory().createPluginInstance();
     if (placementPlugin == null) {
       // Otherwise use the default
-      // TODO: Replace this with a better options, such as the AffinityPlacementFactory
-      placementPlugin = (new SimplePlacementFactory()).createPluginInstance();
+      String defaultPluginId = System.getProperty(DEFAULT_PLACEMENT_PLUGIN_SYSPROP);
+      if (defaultPluginId != null) {
+        switch (defaultPluginId.toLowerCase(Locale.ROOT)) {
+          case "simple":
+            placementPlugin = (new SimplePlacementFactory()).createPluginInstance();
+            break;
+          case "affinity":
+            placementPlugin = (new AffinityPlacementFactory()).createPluginInstance();
+            break;
+          case "minimizecores":
+            placementPlugin = (new MinimizeCoresPlacementFactory()).createPluginInstance();
+            break;
+          case "random":
+            placementPlugin = (new RandomPlacementFactory()).createPluginInstance();
+            break;
+          default:
+            throw new SolrException(
+                SolrException.ErrorCode.SERVER_ERROR,
+                "Invalid value for system property '"
+                    + DEFAULT_PLACEMENT_PLUGIN_SYSPROP
+                    + "'. Supported values are 'simple', 'random', 'affinity' and 'minimizecores'");
+        }
+        log.info(
+            "Default replica placement plugin set in {} to {}",
+            DEFAULT_PLACEMENT_PLUGIN_SYSPROP,
+            defaultPluginId);
+      } else {
+        // TODO: Replace this with a better options, such as the AffinityPlacementFactory
+        placementPlugin = (new SimplePlacementFactory()).createPluginInstance();
+      }
     }
     return new PlacementPluginAssignStrategy(placementPlugin);
```

Comment:
I'm afraid things might get a bit confusing when multiple configurations collide.
There's the default default plugin (Simple), then there's an override to the default (introduced in this PR) and then there's a ZK configured plugin.
It likely makes no practical sense to both have a ZK configured plugin and a default override at the same time. Should we warn here if that's the case?

Reply:
The keyword here is "default". This mechanism is only to set the default plugin. Say you are a platform owner provisioning a Solr cluster for your teams. You set all kind of defaults, like RAM etc, and one of those settings would be the placement policy. Once the teams start using the cluster, they may reconfigure Solr in any way that fits them, thus overriding the pre-packaged defaults.

If we had another mechanism for statically provisioning `/clusterprops.json` we'd likely not have this discussion. Imagine Solr looking for a `$SOLR_HOME/bootstrap/clusterprops.json` at startup, and uploading it to ZK if and only if the file in ZK is empty/blank.

Reply:
I parked this proposal in https://issues.apache.org/jira/browse/SOLR-16493